homeassistant/components/volvooncall/config_flow.py

```diff
@@ -27,64 +33,55 @@ class VolvoOnCallConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """VolvoOnCall config flow."""
 
     VERSION = 1
-    data = None
 
-    async def async_step_user(self, user_input=None) -> FlowResult:
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
         """Handle user step."""
-        errors = {}
+        errors = None
         if user_input is not None:
+            await self.async_set_unique_id(user_input[CONF_USERNAME])
+            self._abort_if_unique_id_configured()
+
             try:
                 await self.is_valid(user_input)
-            except InvalidRegionError:
-                errors[CONF_REGION] = "invalid_region"
-            except AuthenticationError:
-                errors["base"] = "auth"
+            except InvalidAuth:
+                errors = {}
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                errors = {}
+                errors["base"] = "unknown"
```

Comment:
We normally also log the exception stack trace via `LOGGER.exception`.

Please run the scaffold script for an integration config flow to generate best practice code.